experimental/ssh/internal/client/client.go

```diff
@@ -239,6 +241,10 @@ func spawnSSHClient(ctx context.Context, userName, privateKeyPath string, server
 	proxyCommand := fmt.Sprintf("%s ssh connect --proxy --cluster=%s --handover-timeout=%s --metadata=%s,%d --auto-start-cluster=%t",
 		executablePath, opts.ClusterID, opts.HandoverTimeout.String(), userName, serverPort, opts.AutoStartCluster)
 
+	if opts.Profile != "" {
+		proxyCommand += " --profile=" + opts.Profile
+	}
+
```

Comment:
If the goal is to make sure the subprocess inherits the client config, you could leverage `libs/auth/env.go`.